lib/saml_config.rb

```diff
@@ -10,32 +10,37 @@ def settings
       end
-      Hashie::Mash.new(issuer:                        'https://dashboard.login.gov',
-                       idp_sso_target_url:            ENV.fetch('IDP_SSO_URL', saml_env_config[:sso_url]),
-                       idp_slo_target_url:            ENV.fetch('IDP_SLO_URL', saml_env_config[:slo_url]),
-                       idp_cert:                      ENV.fetch('IDP_CERT', saml_env_config[:cert]),
-                       name_identifier_format:        ENV.fetch('IDP_NAME_ID_FORMAT', saml_env_config[:name_id_format]),
-                       certificate:                   ENV.fetch('SP_CERTIFICATE', saml_env_config[:sp_certificate]),
-                       private_key:                   OpenSSL::PKey::RSA.new(
-                         ENV.fetch('SP_PRIVATE_KEY', saml_env_config[:sp_private_key]),
-                         ENV.fetch('SP_PRIVATE_KEY_PASSWORD', saml_env_config[:sp_private_key_password])
-                       ).to_s,
-                       allowed_clock_drift:           5.minutes,
-                       authn_context:                 'http://idmanagement.gov/ns/assurance/loa/1',
-                       single_signon_service_binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST',
-                       double_quote_xml_attribute_values: true,
-                       security: {
-                         authn_requests_signed: true,
-                         embed_sign: true,
-                         digest_method: 'http://www.w3.org/2001/04/xmlenc#sha256',
-                         signature_method: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'
-                       })
+      Hashie::Mash.new(
+        issuer: 'https://dashboard.login.gov',
+        idp_sso_target_url: ENV.fetch('IDP_SSO_URL', saml_env_config[:sso_url]),
+        idp_slo_target_url: ENV.fetch('IDP_SLO_URL', saml_env_config[:slo_url]),
+        idp_cert_fingerprint: ENV.fetch('IDP_FINGERPRINT', saml_env_config[:idp_fingerprint]),
+        name_identifier_format: ENV.fetch('IDP_NAME_ID_FORMAT', saml_env_config[:name_id_format]),
+        certificate: ENV.fetch('SP_CERTIFICATE', saml_env_config[:sp_certificate]),
+        private_key: OpenSSL::PKey::RSA.new(
+          ENV.fetch('SP_PRIVATE_KEY', saml_env_config[:sp_private_key]),
+          ENV.fetch('SP_PRIVATE_KEY_PASSWORD', saml_env_config[:sp_private_key_password])
+        ).to_s,
+        allowed_clock_drift: 5.minutes,
+        authn_context: 'http://idmanagement.gov/ns/assurance/loa/1',
+        single_signon_service_binding: 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST',
+        double_quote_xml_attribute_values: true,
+        security: {
+          authn_requests_signed: true,
+          embed_sign: true,
+          digest_method: 'http://www.w3.org/2001/04/xmlenc#sha256',
+          signature_method: 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'
+        }
+      )
     end
+    # rubocop:enable AbcSize, MethodLength
```

Comment:
I looked at doing it as a constant, it just seemed tricky because I was pulling default values from `saml_env_config` which meant I needed to load that file, and defer it till runtime.